packages/extension-base/src/setup-app.ts

```diff
@@ -17,21 +17,22 @@ import {Config} from './types'
 import withRedBorder from 'overridable!./components/with-red-border'
 import SamplePage from './pages/sample'
 
-const defaultPath = '/sample-page'
 import extensionMeta from '../extension-meta.json'
 
 class Sample extends ApplicationExtension<Config> {
     static readonly id = extensionMeta.id
 
-    extendApp<T>(App: React.ComponentType<T>): React.ComponentType<T> {
+    extendApp<T extends React.ComponentType<T>>(
+        App: React.ComponentType<T>
+    ): React.ComponentType<T> {
         return withRedBorder(App)
     }
 
     extendRoutes(routes: RouteProps[]): RouteProps[] {
         return [
             {
                 exact: true,
-                path: this.getConfig().path || defaultPath,
+                path: this.getConfig().path,
```

Comment:
When an app extension calls its `getConfig`, now it can expect to get the complete config (which has merged the user-defined and default config).